### PR TITLE
fix: region validation avoid NPE

### PR DESCRIPTION
**Pull request by tilo-chen**
**Issue #, if available:**

**Description of changes:**
Properly validate AWS region.

**Why is this change necessary:**
Currently we are letting bad region through and causing NullPointerException later on.

**How was this change tested:**
Added new unit tests.

**Any additional information or context required to review the change:**

**Checklist:**
 - [ ] Updated the README if applicable
 - [x] Updated or added new unit tests
 - [ ] Updated or added new integration tests
 - [ ] Updated or added new end-to-end tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java [hunk 1]
```diff
@@ -127,7 +127,7 @@
     private static final String DEFAULT_ENV_STAGE = "prod";
     private static final String CANNOT_BE_EMPTY = " cannot be empty";
     private static final Logger logger = LogManager.getLogger(DeviceConfiguration.class);
-    private static final String FALLBACK_DEFAULT_REGION = "us-east-1";
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java:130]: make this package private for testing, not public.

File: src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java [hunk 2]
```diff
@@ -470,9 +470,8 @@ private Validator getRegionValidator() {
                     logger.atWarn().log("Error looking up AWS region", ex);
                 }
             }
-            // Snow* devices have a null region
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java:473]: keep the comment.

File: src/test/java/com/aws/greengrass/easysetup/GreengrassSetupTest.java [hunk 3]
```diff
@@ -300,15 +301,25 @@ void GIVEN_blank_user_WHEN_script_is_used_THEN_error() throws Exception {
...
+
+    @Test
+    void GIVEN_setup_script_WHEN_bad_region_provided_THEN_fail() {
+        GreengrassSetup greengrassSetup = new GreengrassSetup(System.out, System.err, deviceProvisioningHelper,
+                platform, kernel, "-i",
+                "mock_config_path", "-r", "mock_root", "-tn", "mock_thing_name", "-trn", "mock_tes_role_name",
+                "-ss", "false", "--aws-region", "nowhere");
+        greengrassSetup.parseArgs();
+        Exception e = assertThrows(RuntimeException.class, greengrassSetup::performSetup);
+        assertTrue(e.getMessage().contains("is invalid AWS region"));
```
Comment (MikeDombo) [inline src/test/java/com/aws/greengrass/easysetup/GreengrassSetupTest.java:322]: use hamcrest. `assertThat(e.getMessage(), containsString(...))`

Comment (tilo-chen) [reply src/test/java/com/aws/greengrass/easysetup/GreengrassSetupTest.java]: Thanks!

**Review by MikeDombo — APPROVED** [review]

**Review by hui-yang — APPROVED** [review]

**Pull request merged**